src/main/java/io/nats/service/api/StatsResponse.java

```diff
@@ -0,0 +1,118 @@
+// Copyright 2022 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package io.nats.service.api;
+
+import io.nats.client.support.JsonSerializable;
+import io.nats.client.support.JsonUtils;
+import io.nats.service.ServiceDescriptor;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import static io.nats.client.support.ApiConstants.*;
+import static io.nats.client.support.JsonUtils.*;
+
+/**
+ * TBD
+ */
+public class StatsResponse implements JsonSerializable {
+    private final String serviceId;
+    private final String name;
+    private final String version;
+    private final List<EndpointStats> stats;
+
+    public StatsResponse(String serviceId, ServiceDescriptor descriptor, EndpointStats stats) {
+        this.serviceId = serviceId;
+        this.name = descriptor.name;
+        this.version = descriptor.version;
+        this.stats = Collections.singletonList(stats);
+    }
+
+    public StatsResponse(String serviceId, ServiceDescriptor descriptor, Collection<EndpointStats> stats) {
+        this.serviceId = serviceId;
+        this.name = descriptor.name;
+        this.version = descriptor.version;
+        this.stats = new ArrayList<>(stats);
+    }
+
+    public StatsResponse(String serviceId, String name, String version, EndpointStats stats) {
+        this.serviceId = serviceId;
+        this.name = name;
+        this.version = version;
+        this.stats = Collections.singletonList(stats);
+    }
+
+    public StatsResponse(String serviceId, String name, String version, Collection<EndpointStats> stats) {
+        this.name = name;
+        this.serviceId = serviceId;
+        this.version = version;
+        this.stats = new ArrayList<>(stats);
+    }
+
+    public StatsResponse(String json) {
+        name = JsonUtils.readString(json, string_pattern(NAME));
+        serviceId = JsonUtils.readString(json, string_pattern("id"));
+        version = JsonUtils.readString(json, VERSION_RE);
```

Comment:
The `_RE` is the regex, aka pattern constant. So I didn't have NAME_RE, ID or ID_RE in the core constants. I probably should add them in as constants, but since this service is in a different package, I was undecided as to where. Probably will end up in the main...